usr/share/rear/backup/NETFS/default/500_make_backup.sh

```diff
@@ -71,45 +71,91 @@ LogPrint "Creating $BACKUP_PROG archive '$backuparchive'"
-			"${BACKUP_PROG_OPTIONS[@]}" $backuparchive \
-			$(cat $TMP_DIR/backup-include.txt) $RUNTIME_LOGFILE > $backuparchive
-	;;
+    # tar compatible programs here
+    (tar)
+        set_tar_features
+        Log $BACKUP_PROG $TAR_OPTIONS --sparse --block-number --totals --verbose \
+            --no-wildcards-match-slash --one-file-system \
+            --ignore-failed-read "${BACKUP_PROG_OPTIONS[@]}" \
+            $BACKUP_PROG_CREATE_NEWER_OPTIONS \
+            ${BACKUP_PROG_BLOCKS:+-b $BACKUP_PROG_BLOCKS} "${BACKUP_PROG_COMPRESS_OPTIONS[@]}" \
+            -X $TMP_DIR/backup-exclude.txt -C / -c -f - \
+            $(cat $TMP_DIR/backup-include.txt) $RUNTIME_LOGFILE \| $BACKUP_PROG_CRYPT_OPTIONS BACKUP_PROG_CRYPT_KEY \| $SPLIT_COMMAND
+
+        # Variable used to record the short name of piped commands in case of
+        # error, e.g. ( "tar" "cat" "dd" ) in case of unencrypted and unsplit backup.
+        backup_prog_shortnames=(
+            "$(basename $(echo "$BACKUP_PROG" | awk '{ print $1 }'))"
+            "$(basename $(echo "$BACKUP_PROG_CRYPT_OPTIONS" | awk '{ print $1 }'))"
+            "$(basename $(echo "$SPLIT_COMMAND" | awk '{ print $1 }'))"
+        )
+        for index in ${!backup_prog_shortnames[@]} ; do
+            [ -n "${backup_prog_shortnames[$index]}" ] || BugError "No computed shortname for pipe component $index"
+        done
+
+        $BACKUP_PROG $TAR_OPTIONS --sparse --block-number --totals --verbose    \
+            --no-wildcards-match-slash --one-file-system                        \
+            --ignore-failed-read "${BACKUP_PROG_OPTIONS[@]}"                    \
+            $BACKUP_PROG_CREATE_NEWER_OPTIONS                                   \
+            ${BACKUP_PROG_BLOCKS:+-b $BACKUP_PROG_BLOCKS}                       \
+            "${BACKUP_PROG_COMPRESS_OPTIONS[@]}"                                \
+            -X $TMP_DIR/backup-exclude.txt -C / -c -f -                         \
+            $(cat $TMP_DIR/backup-include.txt) $RUNTIME_LOGFILE |   \
+                                                                    \
+        $BACKUP_PROG_CRYPT_OPTIONS $BACKUP_PROG_CRYPT_KEY |         \
+                                                                    \
+        $SPLIT_COMMAND
+        pipes_rc=( ${PIPESTATUS[@]} )
+
+        # Exit code logic:
+        # - never return rc=1 (this is reserved for "tar" warning about modified files)
+        # - process exit code in pipe's reverse order
+        #   - if last command failed (e.g. "dd"), return an error
+        #   - otherwise if previous command failed (e.g. "encrypt"), return an error
+        #   ...
+        #   - otherwise return "tar" exit code
+        #
+        # When an error occurs, record the program name in $TMP_DIR/failing_backup_prog
+        # and real exit code in $TMP_DIR/failing_backup_prog_rc.
+
+        let index=${#pipes_rc[@]}-1
+        while [ $index -ge 0 ] ; do
+            rc=${pipes_rc[$index]}
```

Comment:
Only a personal side note:
I wonder if a generic implementation with a computed `index` is really needed
because - as far as I see - the two arrays always contain exactly 3 elements
(if not it is a BugError) so that the code here may rely on it and be simply like
<pre>
for index in 0 1 2 ; do
    ...
done
</pre>
but this is a matter of personal coding style taste.

Reply:
The idea there is that this won't have to be changed if we enhance ReaR.
FYI I've something in the pipe that would require adding a pipe to that tar line.

Reply:
Dear piper,
I am looking forward to you pipe outcome!
;-)